### PR TITLE
Populate TBS and Cert Signature fields in X509Description.

### PR DESCRIPTION
**Pull request by phad**
These fields were already defined but not being populated.
Add 'basic_constraint_ca' field to the X509Description proto as well.

### CODE REVIEW

File: python/ct/client/db/cert_desc.py [hunk 1]
```diff
@@ -48,6 +50,27 @@ def from_cert(certificate, observations=[]):
...
+        proto.cert_signature.algorithm_id = cert_alg.long_name
+
+        cert_params = certificate.signature_algorithm()["parameters"]
+        proto.cert_signature.parameters = str(cert_params)
+    except cert.CertificateError:
+        pass
+
+    try:
+        is_ca = certificate.basic_constraint_ca()
+        proto.basic_constraint_ca = is_ca is not None and is_ca.__bool__()
```
Comment (eranmes) [inline python/ct/client/db/cert_desc.py]: the 'is not None' bit is superfluous. Also, if type(is_ca) is not bool, you'll get an AttributeError here.
Wouldn't you get the same result from this expression by using bool(is_ca) ?

Comment (phad) [inline python/ct/client/db/cert_desc.py]: basic_constraint_ca() can return None, ASN1.Bool (true) or ASN1.Bool(false)
I'm trying to handle each of those.

File: python/ct/client/db/cert_desc.py [hunk 2]
```diff
@@ -48,6 +50,27 @@ def from_cert(certificate, observations=[]):
...
+        proto.cert_signature.algorithm_id = cert_alg.long_name
+
+        cert_params = certificate.signature_algorithm()["parameters"]
+        proto.cert_signature.parameters = str(cert_params)
+    except cert.CertificateError:
+        pass
+
+    try:
+        is_ca = certificate.basic_constraint_ca()
+        proto.basic_constraint_ca = bool(is_ca)
```
Comment (eranmes) [inline python/ct/client/db/cert_desc.py]: Nit: inline is_ca?

File: python/ct/client/db/cert_desc_test.py [hunk 3]
```diff
@@ -8,57 +8,71 @@
...
     def test_from_cert(self):
-        observations = []
-        for check in all_checks.ALL_CHECKS:
-            observations += check.check(CERT) or []
-        observations.append(observation.Observation(
-            "AE", u'ćę©ß→æ→ćąßę-ß©ąńśþa©ęńć←', (u'əę”ąłęµ', u'…łą↓ð→↓ś→ę')))
-        proto = cert_desc.from_cert(CERT, observations)
-        self.assertEqual(proto.der, CERT.to_der())
+        for test_case in [(CERT, False), (CA_CERT, True)]:
+            source = test_case[0]
```
Comment (eranmes) [inline python/ct/client/db/cert_desc_test.py]: Nit: You can unpack both values together:
(source, expect_ca_true) = test_case

**Comment by eranmes** [comment]
LGTM, see two minor comments.

File: python/ct/client/db/cert_desc_test.py (hunk 3 above)
Comment (phad) [inline python/ct/client/db/cert_desc_test.py]: Thanks, I forget about tuple assignment like this.

**Pull request merged**